redfish-core/lib/account_service.hpp

```diff
@@ -1342,6 +1343,71 @@ inline void updateUserProperties(
             }
         }
 
+        if (mfaBypass)
+        {
+            std::string mfaBypassDbusVal;
+            // Check if the input vector contains just one bypass type:
+            // as there are just two defined in the backend:
+            // GoogleAuthenticator  and None
+            if (mfaBypass->size() > 1)
+            {
+                messages::propertyNotWritable(asyncResp->res,
+                                              "MFABypass/BypassTypes");
+                return;
+            }
+            std::string mfaBypassDbusPrefix =
+                "xyz.openbmc_project.User.MultiFactorAuthConfiguration.Type.";
+            if (mfaBypass->empty())
+            {
+                mfaBypassDbusVal = mfaBypassDbusPrefix + "None";
+            }
+            else
+            {
+                std::string mfaBypassVal = mfaBypass->front();
+                if (mfaBypassVal == "GoogleAuthenticator" ||
+                    mfaBypassVal == "None")
+                {
+                    mfaBypassDbusVal = mfaBypassDbusPrefix + mfaBypassVal;
+                }
+                else
+                {
+                    messages::propertyValueNotInList(
+                        asyncResp->res, mfaBypassVal, "MFABypass/BypassTypes");
+                    return;
+                }
+            }
+            sdbusplus::asio::setProperty(
+                *crow::connections::systemBus,
+                "xyz.openbmc_project.User.Manager", dbusObjectPath,
+                "xyz.openbmc_project.User.TOTPAuthenticator",
+                "BypassedProtocol", mfaBypassDbusVal,
+                [asyncResp,
+                 mfaBypassDbusVal](const boost::system::error_code& ec,
+                                   const sdbusplus::message_t& msg) {
+                if (ec)
+                {
+                    const sd_bus_error* dbusError = msg.get_error();
+                    if (dbusError != nullptr)
+                    {
+                        std::string_view errorName(dbusError->name);
+
+                        if (errorName ==
+                            "xyz.openbmc_project.Common.Error.InvalidArgument")
+                        {
+                            BMCWEB_LOG_WARNING("DBUS response error: {}", ec);
+                            messages::propertyValueIncorrect(
+                                asyncResp->res, "MFABypass/BypassTypes",
+                                mfaBypassDbusVal);
+                            return;
+                        }
+                        messages::internalError(asyncResp->res);
```

Comment:
I think moving line 1390 `BMCWEB_LOG_WARNING("DBUS response error: {}", ec);` to this line 1404 
`BMCWEB_LOG_ERROR("DBUS response error: {}", ec); ` makes more sense since it will be on by default

Reply:
So, as per the current code, the warning will be logged followed by what the error is - either propertyValueIncorret or internalError.

This should be okay right?

Reply:
By default, error log level is enabled.
In  case `internalError` error, there'll be no log (about `ec`) in journal.
So, I think L1387-L1403 can be

```
                if (ec)
                {
-                   BMCWEB_LOG_WARNING("DBUS response error: {}", ec);
                    const sd_bus_error* dbusError = msg.get_error();
                    if (dbusError != nullptr)
                    {
                        std::string_view errorName(dbusError->name);

                        if (errorName ==
                            "xyz.openbmc_project.Common.Error.InvalidArgument")
                        {
+                           BMCWEB_LOG_WARNING("DBUS response error: {}", ec);
                            messages::propertyValueIncorrect(
                                asyncResp->res, "MFABypass/BypassTypes",
                                mfaBypassDbusVal);
                            return;
                        }
+                      BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
                        messages::internalError(asyncResp->res);
```